src/main/java/com/internet/shop/model/User.java

```diff
@@ -27,6 +27,9 @@ public User(String name, String login, String password, Set<Role> roles, byte[]
     public User(String user) {
     }
 
+    public User(String name, String login, String password, byte[] salt) {
```

Comment:
there are too many constructors in this class
do you use them all?
let's maybe leave the most relevant ones and set the fields if necessary